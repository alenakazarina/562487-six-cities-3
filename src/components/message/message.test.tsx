import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Message from './message';

const mockFn = jest.fn();

describe(`Message`, () => {
  it(`should render Message correctly`, () => {
    const tree = renderer.create(
        <Message
          status={400}
          onClose={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`should render Message with 401 status`, () => {
    const tree = renderer.create(
        <Message
          status={401}
          onClose={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
