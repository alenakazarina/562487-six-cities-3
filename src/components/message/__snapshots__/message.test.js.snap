// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render Message correctly 1`] = `
<section
  className="message message--error"
  style={
    Object {
      "backgroundColor": "#f47d75",
      "borderRadius": "3px",
      "color": "#ffffff",
      "padding": "20px 20px 10px 20px",
      "position": "fixed",
      "right": "5%",
      "top": "5%",
      "zIndex": 1000,
    }
  }
>
  <div
    className="message__header"
    style={
      Object {
        "borderBottom": "1px solid rgba(255, 255, 255, 0.5)",
        "paddingBottom": "10px",
      }
    }
  >
    <b>
      Error status code: 
      400
    </b>
  </div>
  <div
    className="message__body"
  >
    <p>
      Something went wrong :(
    </p>
    <p>
      Please, try again later.
    </p>
  </div>
  <button
    className="message__button button"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(./img/close.svg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "10px",
        "color": "#ffffff",
        "fontWeight": "bold",
        "padding": "10px",
        "position": "absolute",
        "right": "5px",
        "top": "5px",
        "zIndex": 10,
      }
    }
  />
</section>
`;
